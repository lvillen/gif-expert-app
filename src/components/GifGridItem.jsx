import React from 'react';
import PropTypes from 'prop-types';
import 'animate.css';
import { AddCategory } from './AddCategory';

export const GifGridItem = ({ title, url }) => {
  return (
    <>
      <div className="card animate__animated animate__fadeInUp">
        <img src={ url } alt={ title } />
        <p> { title } </p>
      </div>
    </>
  );
};

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}