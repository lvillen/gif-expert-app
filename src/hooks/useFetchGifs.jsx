import { useState, useEffect } from 'react';
import { getGifs } from '../helpers/getGifs';

export const useFetchGifs = ( categoryProp ) => {
  const [state, setState] = useState({
    data: [],
    loading: true
  });

  useEffect( () => {
    getGifs( categoryProp ).then( imgs => {
      setState({
        data: imgs,
        loading: false
      });
    });
  }, [ categoryProp ]);

  return state;
};