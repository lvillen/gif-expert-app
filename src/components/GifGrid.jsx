import { useState, useEffect } from 'react';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ categoryProp }) => {
  const [images, setImages] = useState([]);

  useEffect( () => {
    getGifs();
  }, []);

  const getGifs = async() => {
    const url = 'https://api.giphy.com/v1/gifs/search?q=Lord+of+the+Rings&limit=10&api_key=wOYIa1qIXGN2erRRDRcJ8KWrqPA6XiyI';
    const response = await fetch(url);
    const { data } = await response.json();

    const gifs = data.map( img => {
      return {
        id: img.id,
        title: img.title,
        url: img.images?.downsized_medium.url
      };
    });

    setImages( gifs );
  }

  return (
    <>
      {console.log(images)}
      <h3>{ categoryProp }</h3>
      {
        images.map( img => (
          <GifGridItem 
            key={img.id} 
            { ...img } 
          />
        ))
      }
    </>
  );
};
