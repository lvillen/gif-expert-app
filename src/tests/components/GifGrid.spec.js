import React from 'react';
import { GifGrid } from '../../components/GifGrid';
import { shallow } from 'enzyme';
import useFetchGifs from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Tests GifGrid Component', () => {
  const category = 'Hello world';
  const mockUseFetchGifs = useFetchGifs;
  
  it('should match the snapshot', () => {
    const wrapper = shallow(<GifGrid categoryProp={ category }/>);

    mockUseFetchGifs.mockImplementation( () => {
      return { data: [], loading: true }
    });

    expect( wrapper ).toMatchSnapshot();
  });

  it('should show items when images loaded with useFetchGifs', () => {
    const wrapper = shallow(<GifGrid categoryProp={ category }/>);
    
    const gifs = [{
      id: 'abc',
      url: 'https://localhost/an/image.jpg',
      title: 'A title'
    }]
    
    useFetchGifs.mockReturnValue({ data: gifs, loading: false });
    
    expect( wrapper ).toMatchSnapshot();
  });
});