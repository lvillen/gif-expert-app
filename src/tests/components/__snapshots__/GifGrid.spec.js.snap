// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests GifGrid Component should match the snapshot 1`] = `
<Fragment>
  <h3>
    Hello world
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
